src/main/java/org/broadinstitute/hellbender/engine/FeatureInput.java

```diff
@@ -70,6 +87,123 @@
+            } catch (final ClassCastException e){
+                throw new UserException("GenomicsDB inputs can only be used to provide VariantContexts.", e);
+            }
+        } else {
+            return getNonGenomicDBFeatureReader();
+        }
+    }
+
+    /**
+     * Helper to organize generating of a GenomicsDB reader
+     */
+    private static class GenomicsDBInput {
+        private final File loaderJson;
+        private final File queryJson;
+
+        public GenomicsDBInput(String path){
+            if( !isGenomeDBPath(path)){
+                throw new GATKException("Trying to create a GenomicsDBReader from a non-genomics db input");
+            }
+
+            final String noheader = path.replace(GENOMIC_DB_URI_SCHEME, "");
+            loaderJson = new File(noheader, "loader.json");
+            queryJson = new File(noheader, "query.json");
+
+            assertJsonExists(loaderJson);
+            assertJsonExists(queryJson);
+        }
+
+        public  FeatureReader<VariantContext> getFeatureReader() {
+            try {
+                return new GenomicsDBFeatureReader<>(loaderJson.getAbsolutePath(), queryJson.getAbsolutePath(), new BCF2Codec());
+            } catch (final IOException e) {
+                throw new GATKException("Couldn't create GenomicsDBFeature reader", e);
+            }
+        }
+
+        private static void assertJsonExists(final File json) {
+            if(!json.exists()) {
+                throw new UserException("Couldn't connect to GenomicsDB because " + json.getAbsolutePath() + " does not exist.");
+            }
+        }
+    }
+
+    private boolean isGenomeDBInput() {
+        return isGenomeDBPath(featureFile);
+    }
+
+    /**
+     * @return readers from this input support fast random access
+     */
+    boolean readersSupportRandomAccess() {
+        return isGenomeDBInput() || nonGenomicsDBFileHasIndex();
+    }
+
+    /**
+     * @return readers created from this input have a separate index file
+     */
+    boolean readersHaveIndex() {
+        return !isGenomeDBInput() && nonGenomicsDBFileHasIndex();
+    }
+
+    private boolean nonGenomicsDBFileHasIndex() {
+        if (hasIndex == null) {
+                try (AbstractFeatureReader<?, ?> reader = getNonGenomicDBFeatureReader()) {
+                    hasIndex = reader.hasIndex();
+                } catch (final IOException e) {
+                    throw new GATKException("Failed to close a reader after looking for an index", e);
+                }
+        }
+        return hasIndex;
+    }
+
+
+
+    private AbstractFeatureReader<T, ?> getNonGenomicDBFeatureReader() {
+        final File featureFile = new File(getFeaturePath());
+        final String absolutePath = featureFile.getAbsolutePath();
+        if (!featureFile.exists()){
+            throw new UserException.CouldNotReadInputFile(featureFile, "It doesn't exist.");
+        }
+        if (!featureFile.isFile()){
```

Comment:
Check `featureFile.canRead()` as well to handle permissions issues (this was previously done in `FeatureDataSource`)